src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/Data.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck.transformation
+
```

Comment:
Maybe would make sense to give a more specific name to this file instead of `Data.kt`?
Something like `LocalVariables.kt`? 

Otherwise, good code here! Nice that now we will have good abstractions/classes to work with local vars.